src/util/http/UrlParser.cpp

```diff
@@ -8,8 +8,9 @@
 
 using namespace ad_utility::url_parser;
 
-std::optional<std::string> ad_utility::url_parser::getParameterCheckAtMostOnce(
-    const ParamValueMap& map, string_view key) {
+const std::optional<std::string>&
```

Comment:
You have to return a value here, the reference is definitely dangling!

Reply:
`optional<string>` was perfectly fine here.

Reply:
Then `Server::checkParameter` will have to return a `optional<string>` instead of `optional<string_view>`. It was returning a dangling reference before which I was trying to fix.